src/v/cloud_storage/topic_manifest.cc

```diff
@@ -41,68 +213,34 @@ ss::future<> topic_manifest::update(ss::input_stream<char> is) {
     co_await ss::copy(is, os);
     iobuf_istreambuf ibuf(result);
     std::istream stream(&ibuf);
-    json::Document m;
+
     rapidjson::IStreamWrapper wrapper(stream);
-    m.ParseStream(wrapper);
-    update(m);
-    co_return;
-}
+    rapidjson::Reader reader;
+    topic_manifest_handler handler;
+    if (reader.Parse(wrapper, handler)) {
+        vlog(cst_log.debug, "Parsed successfully!");
```

Comment:
here is where I think it would make sense to validate the parsed json and check things like the version and value ranges etc...

Reply:
yes, moved version check and lexical cast here